services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -946,61 +949,60 @@ export async function processWorkItems(
 }
 
 /**
- * Update job status/progress in response to a service provided work item update
+ * Group work item updates by its workflow step and return the grouped work item updates
+ * as a map of workflow step to a list of work item updates on that workflow step.
+ * @param updates - List of work item updates
  *
- * @param jobId - job id
- * @param update - information about the work item update
- * @param operation - the DataOperation for the user's request
- * @param logger - the Logger for the request
+ * @returns a map of workflow step to a list of work item updates on that workflow step.
  */
-export async function handleWorkItemUpdateWithJobId(
-  jobID: string,
-  update: WorkItemUpdate,
-  operation: object,
-  logger: Logger,
-  loadNextWorkflowStep = false): Promise<void> {
-  try {
-    let nextWorkflowStep = undefined;
-    if (loadNextWorkflowStep) {
-      nextWorkflowStep = await (await logAsyncExecutionTime(
-        getWorkflowStepByJobIdStepIndex,
-        'HWIUWJI.getWorkflowStepByJobIdStepIndex',
-        logger))(db, jobID, update.workflowStepIndex + 1);
+function groupByWorkflowStepIndex(
+  updates: WorkItemUpdateQueueItem[]): Record<number, WorkItemUpdateQueueItem[]> {
+
+  return updates.reduce((result, currentUpdate) => {
```

Comment:
nit: Map.groupBy is slightly simpler

Reply:
Removed this function and used `groupBy` provided by lodash.